### PR TITLE
fix: mentions not being sent in dev

### LINKED ISSUE #50: Mark mention as read when click navigates to conversation
When clicking on a notification and navigating to the item the notification should be marked as "read"

### LINKED ISSUE #51: Mentions are not sent on message post

### PR DESCRIPTION
**Pull request by spaenleh**
This PR fixes:
- #51: due to a RegEx problem
- #50
- move export and download functionality to the item settings in builder [see this PR](https://github.com/graasp/graasp-compose/pull/426)

closes #51 
closes #50

### CODE REVIEW

File: example/src/components/ChatboxTest.tsx [hunk 1]
```diff
@@ -102,78 +113,95 @@ const ChatboxTest: FC = () => {
     }
   };
 
+  const i18n = useMemo(() => {
+    const i18nInstance = buildI18n(namespaces.chatbox);
+    i18nInstance.changeLanguage(lang);
+    return i18nInstance;
+  }, [lang]);
+
+  console.log(currentMember);
```
Comment (pyphilia) [inline example/src/components/ChatboxTest.tsx:122]: Remove.

File: example/src/components/ChatboxTest.tsx [hunk 2]
```diff
@@ -102,78 +113,95 @@ const ChatboxTest: FC = () => {
...
+            chatId={chatId}
+            lang={lang}
+            showAdminTools={showTools}
+          />
+        </div>
       </div>
-    </div>
+    </I18nextProvider>
   );
 };
```
Comment (pyphilia) [inline example/src/components/ChatboxTest.tsx:206]: I look at this file and wonder if it would match well with storybook? 🤔

File: example/src/components/ChatboxWrapper.tsx [hunk 3]
```diff
@@ -27,15 +26,16 @@ const ChatboxWrapper: FC<Props> = ({
...
   const { data: chat } = hooks.useItemChat(chatId);
-  const memberships = hooks.useItemMemberships(chatId).data;
   // get chat messages
   const chatMessages = chat?.messages;
 
+  const { data: memberships } = hooks.useItemMemberships(chatId);
+
   const memberIds: string[] =
-    memberships?.map((m) => m.memberId)?.toArray() || [];
+    (memberships?.size && memberships?.map((m) => m.memberId)?.toArray()) || [];
```
Comment (pyphilia) [inline example/src/components/ChatboxWrapper.tsx:35]: Did you really need the check on the size? Doesn't the map will be anyway empty and so result in an empty array?

File: example/src/components/ChatboxWrapper.tsx [hunk 4]
```diff
@@ -68,7 +65,7 @@ const ChatboxWrapper: FC<Props> = ({
       sendMessageFunction={sendMessage}
       deleteMessageFunction={deleteMessage}
       editMessageFunction={editMessage}
-      clearChatFunction={clearChat}
+      // clearChatFunction={clearChat}
```
Comment (pyphilia) [inline example/src/components/ChatboxWrapper.tsx:68]: Remove.

File: example/src/config/constants.js [hunk 5]
```diff
@@ -16,5 +16,5 @@ export const TOOLS_SIZE = 64;
 
 export const GRAASP_PANEL_WIDTH = 290;
 
-export const DEFAULT_CHAT_ID = '39370f67-2153-4ab9-9679-b1966542d27d';
+export const DEFAULT_CHAT_ID = '';
```
Comment (pyphilia) [inline example/src/config/constants.js:19]: Maybe you want a fake id here? so it's clearer than an empty string?

File: src/components/Chatbox/EditBanner.tsx [hunk 6]
```diff
@@ -75,7 +75,10 @@ const EditBanner: FC<Props> = ({ onClose, editedText }) => {
             className={classes.oldTextPreview}
             data-cy={editBannerOldTextCypress}
           >
-            {normalizeMentions(editedText)}
+            {
+              // todo: make this work as expected in dev
```
Comment (pyphilia) [inline src/components/Chatbox/EditBanner.tsx:79]: Is it okay now?

File: src/components/Chatbox/Input.tsx [hunk 7]
```diff
@@ -88,7 +89,9 @@ const Input: FC<Props> = ({
       border: '1px solid silver',
       width: '100%',
       overflow: 'auto',
-      height: '70px',
+      // height: '70px',
```
Comment (pyphilia) [inline src/components/Chatbox/Input.tsx:92]: Remove.

File: src/components/Chatbox/Input.tsx [hunk 8]
```diff
@@ -97,7 +100,9 @@ const Input: FC<Props> = ({
       border: '1px solid transparent',
       boxSizing: 'border-box',
       overflow: 'hidden',
-      height: '70px',
+      // height: '70px',
```
Comment (pyphilia) [inline src/components/Chatbox/Input.tsx:103]: Remove.

File: src/components/Chatbox/Input.tsx [hunk 9]
```diff
@@ -148,15 +155,19 @@ const Input: FC<Props> = ({
 
   const onSend = (): void => {
     if (textInput) {
-      const mentions = getAllMentions(textInput).map(({ id }) => id);
-      let expandedMentions: string[] = mentions;
+      let expandedMentions: string[] = currentMentions;
+      console.log('expanded mentions initial', expandedMentions);
```
Comment (pyphilia) [inline src/components/Chatbox/Input.tsx:159]: Remove? use `console.debug`?

File: src/components/Chatbox/Input.tsx [hunk 10]
```diff
@@ -148,15 +155,19 @@ const Input: FC<Props> = ({
...
       // expand '@all' to all members in mentions array (skip if there are no members)
-      if (mentions.includes(ALL_MEMBERS_ID) && members?.size) {
+      if (currentMentions.includes(ALL_MEMBERS_ID) && members?.size) {
         expandedMentions = members.map((m) => m.id).toArray();
+        console.log('All detected, mentions are now', expandedMentions);
       }
+      console.log('Just before sending mentions are', expandedMentions);
```
Comment (pyphilia) [inline src/components/Chatbox/Input.tsx:165]: Remove? Debug?

File: src/components/Chatbox/InputBar.tsx [hunk 11]
```diff
@@ -59,6 +59,7 @@ const InputBar: FC<Props> = ({
         body,
       });
     } else {
+      console.log('message body to send is', body);
```
Comment (pyphilia) [inline src/components/Chatbox/InputBar.tsx:62]: Debug?

**Review by pyphilia — APPROVED** [review]
I've added minor suggestions, nice debugging! 🦧

File: example/src/components/ChatboxTest.tsx (hunk 2 above)
Comment (spaenleh) [reply example/src/components/ChatboxTest.tsx:206]: @pyphilia indeed you are right, it would match well with storybook to test the different props and parameters and then be able to interact with it. I will open an issue to tackle that.

File: example/src/components/ChatboxWrapper.tsx (hunk 3 above)
Comment (spaenleh) [reply example/src/components/ChatboxWrapper.tsx:35]: When i tested it was not behaving as expected. Adding the test on size made sure the `List` was non-empty (`.isEmpty()` did not work ...)
